### PR TITLE
Add configuration by profile to AdvanceConfigurationExample [18694]

### PR DESCRIPTION
**Pull request by jparisu**
Add a new argument to AdvanceConfigurationExample that allows to configure a DomainParticipant from XML.

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- [N/A] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- [N/A] Any new/modified methods have been properly documented using Doxygen. 
- [x] Changes are ABI compatible. 
- [x] Changes are API compatible. 
- [N/A] New feature has been added to the `versions.md` file (if applicable).
- [N/A] New feature has been documented/Current behavior is correctly described in the documentation. 
    
- [N/A] Applicable backports have been included in the description.

## Reviewer Checklist
- [x] The PR has a milestone assigned.
- [x] Check contributor checklist is correct.
- [N/A] Check CI results: changes do not issue any warning.
- [N/A] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: examples/cpp/dds/AdvancedConfigurationExample/arg_configuration.h [hunk 1]
```diff
@@ -229,6 +230,8 @@ const option::Descriptor usage[] = {
       "  -p <str> \t--partitions=<str> \tPartitions to match separated by ';'."
       " Single or double quotes required with multiple partitions."
       " With empty string ('') no partitions used. (Default: '')." },
+    { PROFILE, 0, "x", "profile",         Arg::String,
+      "  -x <str> \t--xml-profile \tProfile name to configure DomainParticipant."},
```
Comment (JLBuenoLopez-eProsima) [inline examples/cpp/dds/AdvancedConfigurationExample/arg_configuration.h:234]: The argument parser recognizes the `--profile` option and not the `--xml-profile` as it is printed. Please fix this inconsistency.

**Review by JLBuenoLopez-eProsima — CHANGES_REQUESTED** [review]
1. I think some changes in your local repository have made it to this PR. The commit and PR title states that the goal is to introduce XML configuration support in the AdvanceConfigurationExample, but the SecureHelloWorldExample and the HelloWorldExample are also modified and I think that was not your intention.
1. Please, fix the uncrustify linter which is complaining in the AdvanceConfigurationExample main file.
1. README has not been updated with this new option.
1. Being an example, and thus oriented for user understanding, maybe we should add an XML configuration file so the user can check this functionality.

**Comment by jparisu** [comment]
1. Fixed in https://github.com/eProsima/Fast-DDS/pull/3554/commits/626df5a444c07759ae117f4e891405b5f33976c2
2. https://github.com/eProsima/Fast-DDS/pull/3554/commits/7eed7f62aa7d79a333163ca53f4c20c6c024751e
3. https://github.com/eProsima/Fast-DDS/pull/3554/commits/1b36e045d40346d138f5630c71e2203e80322920
4. https://github.com/eProsima/Fast-DDS/pull/3554/commits/1b36e045d40346d138f5630c71e2203e80322920
5. :+1:

**Comment by JesusPoderoso** [comment]
@richiprosima please test this

File: examples/cpp/dds/AdvancedConfigurationExample/README.md [hunk 2]
```diff
@@ -83,3 +85,16 @@ Discovery options:
                     Increase it to avoid discovery issues on scenarios with
                     several routers. Maximum: 255.
 ```
+
+### XML Configuration
+
+Using argument `--xml-profile <profile_name>` will configure the internal DomainParticipant using the profile name loaded from an XML file.
+To load XML files check [following documentation](https://fast-dds.docs.eprosima.com/en/latest/fastdds/xml_configuration/xml_configuration.html).
```
Comment (JLBuenoLopez-eProsima) [inline examples/cpp/dds/AdvancedConfigurationExample/README.md:92]:
```suggestion
To load XML files check [Fast DDS documentation](https://fast-dds.docs.eprosima.com/en/latest/fastdds/xml_configuration/xml_configuration.html).
```

File: examples/cpp/dds/HelloWorldExample/HelloWorldPublisher.cpp [hunk 3]
```diff
@@ -54,7 +54,7 @@ bool HelloWorldPublisher::init(
         factory->get_default_participant_qos(pqos);
     }
 
-    participant_ = factory->create_participant(0, pqos);
+    participant_ = factory->create_participant(201, pqos);
```
Comment (JLBuenoLopez-eProsima) [inline examples/cpp/dds/HelloWorldExample/HelloWorldPublisher.cpp:57]: This should not be here :smiling_face_with_tear:

**Review by JLBuenoLopez-eProsima — CHANGES_REQUESTED** [review]

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please test this

**Pull request merged**